server/routes/curfew.js

```diff
@@ -1,15 +1,87 @@
 const { asyncMiddleware } = require('../utils/middleware')
 const createStandardRoutes = require('./routeWorkers/standard')
 const { getPathFor } = require('../utils/routes')
-const { getIn } = require('../utils/functionalHelpers')
+const { getIn, mergeWithRight } = require('../utils/functionalHelpers')
 const formConfig = require('./config/curfew')
 
-module.exports = ({ licenceService }) => (router, audited) => {
+module.exports = ({ licenceService, nomisPushService }) => (router, audited, { pushToNomis }) => {
   const standard = createStandardRoutes({ formConfig, licenceService, sectionName: 'curfew' })
 
   router.get('/approvedPremises/:bookingId', addressReviewGets('approvedPremises'))
   router.get('/approvedPremises/:action/:bookingId', addressReviewGets('approvedPremises'))
 
+  router.get('/approvedPremisesChoice/:action/:bookingId', asyncMiddleware(getChoice))
+  router.get('/approvedPremisesChoice/:bookingId', asyncMiddleware(getChoice))
+
+  function getChoice(req, res) {
+    const { bookingId } = req.params
+    const { licence } = res.locals
+    const data = { decision: getApprovedPremisesChoice(getIn(licence, ['licence'])) }
+    const viewData = { data, errorObject: {}, bookingId }
+
+    return res.render('curfew/approvedPremisesChoice', viewData)
+  }
+
+  function getApprovedPremisesChoice(licence) {
+    if (isYes(licence, ['proposedAddress', 'optOut', 'decision'])) {
+      return 'OptOut'
+    }
+
+    if (isYes(licence, ['curfew', 'approvedPremises', 'required'])) {
+      return 'ApprovedPremises'
+    }
+
+    return null
+  }
+
+  function isYes(licence, pathSegments) {
+    const answer = getIn(licence, pathSegments)
+    return answer && answer === 'Yes'
+  }
```

Comment:
all the above is as per similar routes elsewhere - get the data out of the licence record and adapt for display